src/NHibernate/Linq/ReWriters/ConditionalQueryReferenceExpander.cs

```diff
@@ -77,7 +77,7 @@ public void Transform(ResultOperatorBase resultOperator)
 			protected override Expression VisitMember(MemberExpression node)
 			{
 				var result = (MemberExpression) base.VisitMember(node);
-				if (QueryReferenceCounter.CountReferences(result.Expression) > 1)
+				if (result.Expression != null && QueryReferenceCounter.CountReferences(result.Expression) > 1)
```

Comment:
This additional change, required since #1880 is merged, is not a regression from #1880 since other places in the Linq provider were failing for the same reason in case of non-pre-evaluated static property and are also fixed by the db-side method PR. So it is also fixed here.